perfkitbenchmarker/linux_benchmarks/tensorflow_benchmark.py

```diff
@@ -0,0 +1,234 @@
+flags.DEFINE_boolean('tf_use_nccl', True,
+                     'Whether to use nccl all-reduce primitives where possible')
+flags.DEFINE_boolean('tf_distortions', True,
+                     '''Enable/disable distortions during image preprocessing.
+                     These include bbox and color distortions.''')
+
+
+class TFParseOutputException(Exception):
+  pass
+
+
+def GetConfig(user_config):
+  """Load and return benchmark config.
+
+  Args:
+    user_config: user supplied configuration (flags and config file)
+
+  Returns:
+    loaded benchmark configuration
+  """
+  return configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+
+
+def _UpdateBenchmarkSpecWithFlags(benchmark_spec):
+  """Update the benchmark_spec with supplied command line flags.
+
+  Args:
+    benchmark_spec: benchmark specification to update
+  """
+  benchmark_spec.model = FLAGS.tf_model
+  benchmark_spec.data_name = FLAGS.tf_data_name
+  benchmark_spec.batch_size = FLAGS.tf_batch_size
+  benchmark_spec.variable_update = FLAGS.tf_variable_update
+  benchmark_spec.local_parameter_device = FLAGS.tf_local_parameter_device
+  benchmark_spec.use_nccl = FLAGS.tf_use_nccl
+  benchmark_spec.distortions = FLAGS.tf_distortions
+
+
+def Prepare(benchmark_spec):
+  """Install and set up TensorFlow on the target vm.
+
+  Args:
+    benchmark_spec: The benchmark specification
+  """
+  _UpdateBenchmarkSpecWithFlags(benchmark_spec)
+  vms = benchmark_spec.vms
+  master_vm = vms[0]
+  logging.info('Installing CUDA Toolkit 8.0 on %s', master_vm)
+  master_vm.Install('cuda_toolkit_8')
+  benchmark_spec.num_gpus = cuda_toolkit_8.QueryNumberOfGpus(master_vm)
+  master_vm.Install('cudnn')
+  master_vm.Install('tensorflow')
+
+
+def _CreateMetadataDict(benchmark_spec):
+  """Create metadata dict to be used in run results.
+
+  Args:
+    benchmark_spec: benchmark spec
+
+  Returns:
+    metadata dict
+  """
+  metadata = dict()
+  metadata.update(cuda_toolkit_8.GetMetadataFromFlags())
+  metadata['model'] = benchmark_spec.model
+  metadata['num_gpus'] = benchmark_spec.num_gpus
+  metadata['data_name'] = benchmark_spec.data_name
+  metadata['batch_size'] = benchmark_spec.batch_size
+  metadata['variable_update'] = benchmark_spec.variable_update
+  metadata['local_parameter_device'] = benchmark_spec.local_parameter_device
+  metadata['use_nccl'] = benchmark_spec.use_nccl
+  metadata['distortions'] = benchmark_spec.distortions
+  return metadata
+
+
+def _GetEnvironmentVars(vm):
+  """Return a string containing TensorFlow-related environment variables.
+
+  Args:
+    vm: vm to get environment varibles
+
+  Returns:
+    string of environment variables
+  """
+
+  output, _ = vm.RemoteCommand('getconf LONG_BIT', should_log=True)
+  long_bit = output.strip()
+  lib_name = 'lib' if long_bit == '32' else 'lib64'
+  return ' '.join([
+      'PATH=%s' % os.path.join(CUDA_TOOLKIT_INSTALL_DIR,
+                               'bin${PATH:+:${PATH}}'),
```

Comment:
What is going on here? Sorry, I don't understand this syntax.  If you are trying to add the cuda bin dir to path, maybe do this:
```
PATH='os.environ.get('PATH'):%s' % (
    os.path.join(CUDA_TOOLKIT_INSTALL_DIR, 'bin'))
```

If you are trying to do something else, can you explain? Thanks :)